WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -0,0 +1,202 @@
+			}
+		}
+
+		public async Task StartMixingCoinsAsync()
+		{
+			await StartAsync(DisposeCts.Token).ConfigureAwait(false);
+		}
+
+		private async Task<AliceClient[]> RegisterCoinsAsync(CancellationToken stoppingToken)
+		{
+			var aliceArenaClient = new ArenaClient(
+				Round.AmountCredentialIssuerParameters,
+				Round.VsizeCredentialIssuerParameters,
+				AmountCredentialPool,
+				VsizeCredentialPool,
+				ArenaRequestHandler,
+				SecureRandom);
+
+			List<AliceClient> aliceClients = new();
+
+			foreach (var coin in Coins)
+			{
+				var secret = Keymanager.GetSecrets(Kitchen.SaltSoup(), coin.ScriptPubKey.WitHash.ScriptPubKey).First().PrivateKey.GetBitcoinSecret(Keymanager.GetNetwork());
+				aliceClients.Add(await AliceClient.CreateNewAsync(aliceArenaClient, coin, secret, Round.Id, Round.FeeRate).ConfigureAwait(false));
+				await Task.Delay(Random.Next(0, 1000), stoppingToken).ConfigureAwait(false);
+			}
+
+			return aliceClients.ToArray();
+		}
+
+		private async Task ConfirmConnectionsAsync(IEnumerable<AliceClient> aliceClients, CancellationToken stoppingToken)
+		{
+			foreach (var alice in aliceClients)
+			{
+				await alice.ConfirmConnectionAsync(TimeSpan.FromMilliseconds(Random.Next(1000, 5000)), stoppingToken).ConfigureAwait(false);
+				await Task.Delay(Random.Next(0, 1000), stoppingToken).ConfigureAwait(false);
+			}
+		}
+
+		private async Task ReissueAndRegisterOutputsAsync(IEnumerable<(Money Amount, HdPubKey Pubkey)> outputs, CancellationToken stoppingToken)
+		{
+			ArenaClient bobArenaClient = new(
+				Round.AmountCredentialIssuerParameters,
+				Round.VsizeCredentialIssuerParameters,
+				AmountCredentialPool,
+				VsizeCredentialPool,
+				ArenaRequestHandler,
+				SecureRandom);
+
+			BobClient bobClient = new(Round.Id, bobArenaClient);
+
+			Money remaining = outputs.Sum(o => o.Amount);
+
+			var remainingAmountCredentials = AmountCredentialPool.Valuable.Single();
+			var remainingVsizeCredentials = VsizeCredentialPool.Valuable.Single();
+
+			foreach (var output in outputs)
+			{
+				var justNeedtheSize = output.Pubkey.PubKey.WitHash.ScriptPubKey;
+				remaining -= output.Amount;
+
+				var result = await bobArenaClient.ReissueCredentialAsync(
+					Round.Id,
+					output.Amount,
+					output.Pubkey.PubKey.WitHash.ScriptPubKey,
+					remaining,
+					justNeedtheSize,
+					new[] { remainingAmountCredentials },
+					new[] { remainingVsizeCredentials }).ConfigureAwait(false);
+
+				remainingAmountCredentials = result.RealAmountCredentials.Last();
+				remainingVsizeCredentials = result.RealVsizeCredentials.Last();
+
+				await bobClient.RegisterOutputAsync(
+					output.Amount,
+					output.Pubkey.PubKey.WitHash.ScriptPubKey,
+					new[] { result.RealAmountCredentials.First() },
+					new[] { result.RealVsizeCredentials.First() }).ConfigureAwait(false);
+
+				await Task.Delay(Random.Next(0, 1000), stoppingToken).ConfigureAwait(false);
```

Comment:
`Delay` should be in the `WabiSabiApiClient` because this delay is not a business logic but a network/communication randomized delay for privacy.

Reply:
i disagree with both the code and your comment ;-)

i think it's out of scope for this PR regardless of how we choose to do it, but we should talk about it in the next call